File: result/translate/starchat/java2python/python/LENGTH_LONGEST_SUB_STRING_CAN_MAKE_REMOVED.py
def f_gold(str):
        arr = [(" ", -1)]
        maxlen = 0
        for i in range(len(str)):
            arr.append((str[i], i))
            while len(arr) >= 3 and arr[-3][0] == "1" and arr[-2][0] == "0" and arr[-1][0] == "0":
                arr.pop()
            tmp = arr[-1][1]
            maxlen = max(maxlen, i - tmp)
        return maxlen

File: result/translate/chatgpt/java2cpp/c++/LENGTH_LONGEST_SUB_STRING_CAN_MAKE_REMOVED.cpp
#include <vector>
#include <utility>
#include <algorithm>

int f_gold(std::string str) {
    std::vector<std::pair<char, int>> arr;
    arr.push_back(std::make_pair('@', -1));
    int maxlen = 0;
    for (int i = 0; i < str.length(); ++i) {
        arr.push_back(std::make_pair(str.at(i), i));
        while (arr.size() >= 3 && arr[arr.size() - 3].first == '1' && arr[arr.size() - 2].first == '0' && arr[arr.size() - 1].first == '0') {
            arr.erase(arr.end()-3, arr.end());
        }
        int tmp = arr[arr.size() - 1].second;
        maxlen = std::max(maxlen, i - tmp);
    }
    return maxlen;
}